File: Level 0/코딩 기초 트레이닝/조건에 맞게 수열 변환하기 2/solution.js
function solution(arr) {
  let answer = 0;
  let before = [];

  while (JSON.stringify(before) !== JSON.stringify(arr)) {
    before = [...arr];

    arr = arr.map((num) => {
      if (num >= 50 && !(num % 2)) return num / 2;
      if (num < 50 && num % 2) return num * 2 + 1;
      return num;
    });

    answer++;
  }

  return answer - 1;
}

File: Level 0/코딩 기초 트레이닝/조건에 맞게 수열 변환하기 2/solution.py
def solution(arr):
    answer = 0
    before = []

    while before != arr:
        before = arr.copy()

        for i, num in enumerate(arr):
            if num >= 50 and not num % 2:
                arr[i] = num / 2
            elif num < 50 and num % 2:
                arr[i] = num * 2 + 1
            else:
                arr[i] = num

        answer += 1

    return answer - 1
